### ISSUE TITLE
joint.util.breaktext incorrectly measures height

### ISSUE THREAD

**Issue by 40rn05lyv** [clientIO/joint#346]
Line height in JointJs is calculated as:

```
// get line height as text height / 0.8 (as text height is approx. 0.8em
// and line height is 1em. See vectorizer.text())
var lh = lh || textElement.getBBox().height * 1.25;
```

The comment tells that we need to increase line height, because of `vectorizer.text()`. However, if you try to use this function, you'll see that text height is far lower than size:

```
var graph = new joint.dia.Graph;
var paper = new joint.dia.Paper({
  el: $('#paper'),
  width: 1600,
  height: 600,
  model: graph,
  gridSize: 1
});
var size = { width: 50, height: 250 };
var text = joint.util.breakText("This is very very very very very very very very very very very very very very very very very very very very very very long text", size)
var cell = new joint.shapes.basic.Rect({
  size: size,
  attrs: {
    text: {
      text: text
    }
  }
});
graph.addCell(cell);
```

**Comment by 40rn05lyv** [comment]
Even if I increase height for `breakText` function by 1.25 it still doesn't work properly, because `breakText` measures `lh` only for `<text><tspan></tspan</text>` and then multiply it on number of lines.

However, in reality `<text><tspan></tspan</text>.getBBox().height*2` is NOT equal to `<text><tspan></tspan<tspan></tspan</text>`.getBBox().height`.

**Comment by 40rn05lyv** [comment]
Can be fixed with something like this:

```
var oldBreakText = joint.util.breakText;
joint.util.breakText = function(text, size, styles, opt) {
  var height = size.height;

  // Call old break text, but without height
  var brokenText = oldBreakText(text, { width: size.width }, styles, opt);
  var lines = brokenText.split('\n');

  opt = opt || {};
  var svgDocument = opt.svgDocument || V('svg').node;
  var textElement = V('<text></text>').attr(styles || {}).node;
  textElement.style.display = 'block';
  svgDocument.appendChild(textElement);
  if (!opt.svgDocument) {
    document.body.appendChild(svgDocument);
  }

  for (var i = 0; i < lines.length; i++) {
    var textSpan = V('<tspan></tspan>').attr({ x: 0, dy: '1em' }).node;
    var textNode = document.createTextNode('');
    textNode.data = lines[i];
    textSpan.appendChild(textNode);
    textSpan.style.display = 'block';
    textElement.appendChild(textSpan);
    if (textElement.getBBox().height > height) {
      // remove overflowing lines
      lines.splice(i);
      break;
    }
  }

  if (opt.svgDocument) {
    // svg document was provided, remove the text element only
    svgDocument.removeChild(textElement);
  } else {
    // clean svg document
    document.body.removeChild(svgDocument);
  }

  return lines.join('\n');
}
```

**Comment by kumilingus** [comment]
Fixed by #1972.
